### PR TITLE
test: Follow up on Spark 3.4 diff

### PR DESCRIPTION
**Pull request by sunchao**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Certain changes to address comments from https://github.com/apache/arrow-datafusion-comet/pull/166 were somehow excluded in the final commit. This adds them back.

### CODE REVIEW

**Comment by sunchao** [comment]
cc @viirya

File: dev/diffs/3.4.2.diff [hunk 1]
```diff
@@ -361,6 +316,54 @@ index 2796b1cf154..94591f83c84 100644
            }.flatten
            assert(filters.contains(GreaterThan(scan.logicalPlan.output.head, Literal(5L))))
          }
+diff --git a/sql/core/src/test/scala/org/apache/spark/sql/IgnoreComet.scala b/sql/core/src/test/scala/org/apache/spark/sql/IgnoreComet.scala
+new file mode 100644
+index 00000000000..4b31bea33de
+--- /dev/null
++++ b/sql/core/src/test/scala/org/apache/spark/sql/IgnoreComet.scala
```
Comment (viirya) [inline dev/diffs/3.4.2.diff:323]: This is main change? `DisableCometSuite` -> `IgnoreCometSuite`?

File: dev/diffs/3.4.2.diff [hunk 2]
```diff
@@ -976,25 +980,18 @@ index 266bb343526..85ec36db996 100644
...
          // check existence of shuffle
          assert(
 -          joinOperator.left.exists(_.isInstanceOf[ShuffleExchangeExec]) == shuffleLeft,
-+          joinOperator.left.find { op =>
-+            op.isInstanceOf[SortExec] ||
-+              (op.isInstanceOf[CometExec] &&
-+                op.asInstanceOf[CometExec].originalPlan.find(_.isInstanceOf[SortExec]).isDefined)
-+          }.isDefined == sortLeft,
-+
++          joinOperator.left.exists(op => op.isInstanceOf[ShuffleExchangeExec] ||
```
Comment (sunchao) [inline dev/diffs/3.4.2.diff:988]: here's another change

Comment (viirya) [reply dev/diffs/3.4.2.diff:988]: For `op.isInstanceOf[ShuffleExchangeExec]`, don't we need to check if it equals to `shuffleLeft`?

Comment (sunchao) [reply dev/diffs/3.4.2.diff:988]: hmm? it is checked in the code: 

```scala
exists(op => op.isInstanceOf[ShuffleExchangeExec] || op.isInstanceOf[CometShuffleExchangeExec]) == shuffleLeft

Comment (viirya) [reply dev/diffs/3.4.2.diff:988]: Oh, yea, I read it wrongly.

File: dev/diffs/3.4.2.diff [hunk 3]
```diff
@@ -976,25 +980,18 @@ index 266bb343526..85ec36db996 100644
...
+@@ -461,18 +472,22 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils with Adapti
  
          // check existence of shuffle
          assert(
 -          joinOperator.left.exists(_.isInstanceOf[ShuffleExchangeExec]) == shuffleLeft,
-+          joinOperator.left.find { op =>
-+            op.isInstanceOf[SortExec] ||
-+              (op.isInstanceOf[CometExec] &&
-+                op.asInstanceOf[CometExec].originalPlan.find(_.isInstanceOf[SortExec]).isDefined)
-+          }.isDefined == sortLeft,
```
Comment (viirya) [inline dev/diffs/3.4.2.diff:988]: I forgot why we changed to check sort instead of shuffle here.

**Review by viirya — APPROVED** [review]

**Comment by sunchao** [comment]
Thanks, merged

**Pull request merged**